### PR TITLE
Allow underscores in environment variable names

### PR DESCRIPTION
**Pull request by cechandler**
Small addition to allow environment variables to contain underscore characters

### CODE REVIEW

File: grammars/supercollider.cson [hunk 1]
```diff
@@ -69,7 +69,7 @@
     'match': '^\\s*(\\**[a-z]{1}[a-zA-Z0-9_]+)\\s*\\{'
   }
   {
-    'match': '\\~[a-zA-Z0-9]+'
+    'match': '\\~[a-zA-Z0-9_]+'
```
Comment (mossheim) [inline grammars/supercollider.cson]: thanks! this is close, but not correct. `~_` isn't a valid environment variable name, nor is `~_a`. i think it should be `\\~[a-zA-Z0-9][a-zA-Z0-9_]*` then, right? (not 100% sure this notation works like the regex dialect i'm familiar with). the Supercollider IDE highlights `~_a`, i think that ought to be emended

**Review by mossheim — CHANGES_REQUESTED** [review]

File: grammars/supercollider.cson (hunk 1 above)
Comment (telephon) [reply grammars/supercollider.cson]: also something like `~900` isn't valid. So maybe something like

`\\~[a-zA-Z][a-zA-Z0-9_]*`
?

Comment (mossheim) [reply grammars/supercollider.cson]: Oh wow, actually `~ABC` isn't valid either! I learned something new today. So really,

`\\~[a-z][a-zA-Z0-9_]*`

Comment (cechandler) [reply grammars/supercollider.cson]: Ah, great catches. This was a quick fix while I was working this morning, but this is more robust. I tested your solution, and it correctly does not highlight `~_`, `~_a`, `~ABC`, or `~900`.

Should I amend it and resubmit with `\\~[a-z][a-zA-Z0-9_]*`?

Comment (mossheim) [reply grammars/supercollider.cson]: Yes please, thank you!

**Review by telephon — APPROVED** [review]

**Review by mossheim — APPROVED** [review]
Thanks!

**Comment by crucialfelix** [comment]
@brianlheim The atom release isn't that important because supercollider is the package that is mainly used (and I've merged the same changes in over there)

But to do that release you (or probably I) have to do `apm publish patch` which will create the `v.0.2.2` tag, push to github and push to atom's package repo all at once. So the tag you created here shouldn't have been done. 

But it's really not important. just FYI for future

thanks !

**Comment by mossheim** [comment]
thank you for the explanation @crucialfelix! The readme doesn't have any instructions for maintainers, so I thought it was as simple as creating a tag on GH. that would be a good process to document.

**Comment by mossheim** [comment]
@crucialfelix , would you like me to delete the release I made for cleanliness' sake?

**Comment by crucialfelix** [comment]
I'll edit the readme

**Pull request merged**